tests/web/test_check_clickhouse.py

```diff
@@ -88,3 +88,16 @@ def test_dataset_undefined_storage_set(
     assert not check_clickhouse(ignore_experimental=True, metric_tags=metrics_tags)
     for v in metrics_tags.values():
         assert isinstance(v, str)
+
+
+@mock.patch(
+    "snuba.web.views.get_enabled_dataset_names",
+    return_value=["events", "experimental"],
+)
+@mock.patch("snuba.web.views.get_dataset", side_effect=fake_get_dataset)
+@pytest.mark.clickhouse_db
+def test_filter_checked_storages(mock1: mock.MagicMock, mock2: mock.MagicMock) -> None:
```

Comment:
does this test actually test your changes? I don't see anything about readiness states in here

Reply:
It didn't initially because I thought the test could be simplified when we eventually remove `is_experimental`. But, it's better to have a test for it even when this code isn't used yet. Added!